geoextent/lib/extent.py

```diff
@@ -0,0 +1,129 @@
+
+import sys
+import os
+import getopt
+import datetime
+import threading
+import logging
+
+import geoextent.lib.handleCSV as handleCSV
+import geoextent.lib.handleGeojson as handleGeojson
+import geoextent.lib.handleShapefile as handleShapefile
+import geoextent.lib.handleGeotiff as handleGeotiff
+import geoextent.lib.helpfunctions as hf
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+def computeBboxInWGS84(module, path):
+    ''' 
+    input "module": type module, module from which methods shall be used \n
+    input "path": type string, path to file \n
+    returns a bounding box, type list, length = 4 , type = float, schema = [min(longs), min(lats), max(longs), max(lats)], the boudning box has either its original crs or WGS84 (transformed).
+    '''
+    bbox_in_orig_crs = module.getBoundingBox(path)
+    try:
+        #TODO: Add function using to reproject coordinates system
+        if module.fileType == "application/shp":
+            crs = 'None'
+            return bbox_in_orig_crs
+        else:
+            crs = module.getCRS(path)
+    except:
+        pass
+    if 'crs' in locals() and crs and bbox_in_orig_crs:
+        bbox_transformed = hf.transformingArrayIntoWGS84(crs, bbox_in_orig_crs)
+        return bbox_transformed
+    else:
+        raise Exception("The bounding box could not be related to a CRS")
+
+
+def fromFile(filePath, whatMetadata):
+    ''' function is called when filePath is included in commandline (with tag 'b')
+    how this is done depends on the file format - the function calls the extractMetadataFrom<format>()-function \n
+    input "filePath": type string, path to file from which the metadata shall be extracted \n
+    input "whatMetadata": type string, specifices which metadata should be extracted  \n
+    returns None if the format is not supported, else returns the metadata of the file as a dict 
+    (possible) keys of the dict: 'temporal_extent', 'bbox', 'vector_reps', 'crs'
+    '''
+    logging.info("Extracting {} from file {}\n".format(whatMetadata, filePath))
+    
+    fileFormat = filePath[filePath.rfind('.')+1:]
```

Comment:
`filePath.rfind()` is quite error prone here... what about file formats like `tar.gz`?

_Never ever_ handle file paths as strings, but always with the Python API for paths, in this case https://docs.python.org/3/library/os.path.html#os.path.splitext: `filename, extension = os.path.splitext(filePath)
`

Reply:
I did not think about (tar.gz) ;)

Reply:
regarding `tar.gz` it works the same using splitext or r.finder. As the last returns the heighest index of '.' in path.

Reply:
Done.